storage/uri.go

```diff
@@ -211,6 +211,42 @@ func Delete(u fyne.URI) error {
 
 }
 
+func DeleteRecursive(u fyne.URI) error {
+	list, err := List(u)
+
+	if err != nil {
+		return err
+	}
+
+	for _, v := range list {
+		listable, err := CanList(v)
+		if err != nil {
+			return err
+		}
+		if !listable {
+			err = Delete(v)
+			if err != nil {
+				return err
+			}
+			continue
+		} else {
+			err = DeleteRecursive(v)
+
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
You can remove the else statement due to the continue call.
```suggestion
			continue
		}

		err = DeleteRecursive(v)
		if err != nil {
			return err
		}
```